oci/spec_opts_unix.go

```diff
@@ -96,21 +96,21 @@ func WithImageConfig(image Image) SpecOpts {
 		cmd := config.Cmd
 		s.Process.Args = append(config.Entrypoint, cmd...)
 		if config.User != "" {
+			// According to OCI Image Spec v1.0.0, the following are valid for Linux:
+			//   user, uid, user:group, uid:gid, uid:group, user:gid
 			parts := strings.Split(config.User, ":")
 			switch len(parts) {
 			case 1:
 				v, err := strconv.Atoi(parts[0])
 				if err != nil {
 					// if we cannot parse as a uint they try to see if it is a username
-					if err := WithUsername(config.User)(ctx, client, c, s); err != nil {
-						return err
-					}
-					return err
```

Comment:
I'm not sure about this change, I think the old code may have been wrong as well.

If `WithUsername` is successful, shouldn't it continue to run the code after the switch (setting cwd) ?

Alternatively may this can be refactored so that the CWD code is above this switch, which would let us keep this early return, and also let us `return WithUserID(...)`  below, instead of checking `err` there as well.

Reply:
fixed, thanks